src/dbnode/persist/schema/version_checker.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package schema
+
+// VersionChecker centralizes logic for checking if a major, minor version combo supports
+// specific functionality
+type VersionChecker struct {
+	majorVersion int
+	minorVersion int
+}
+
+// NewVersionChecker creates a new VersionChecker
+func NewVersionChecker(majorVersion int, minorVersion int) *VersionChecker {
```

Comment:
This could be a non-pointer type yes? It's just an immutable data container as far as can see? 

Might be worth avoiding the heap allocation if so.

Reply:
> This could be a non-pointer type yes? It's just an immutable data container as far as can see?

That's correct.

> Might be worth avoiding the heap allocation if so.

For future reference, how should one be thinking about managing heap allocations vs data duplication (in this case it's trivial, but perhaps that's different in a future case)?

Reply:
It's a tough one, I would opt where it's easy to make guaranteed non-heap structures with by value structs it's probably worth doing. 

You can also check if something "escapes" or not here:
https://kuree.gitbooks.io/the-go-programming-language-report/content/27/text.html